### DISCUSSION TITLE
Handling the operation names

### DISCUSSION THREAD

**Discussion by webmiche** [xdslproject/xdsl#197] [Ideas]
Currently, our operations have a dedicated field called `name` where we store their name for printing/parsing. However, this means that the `name` field cannot be used for Attributes/Regions/Operands...

This becomes particularly hard in the context of representing MLIR dialects, some of which actually have a `name` attribute (in IRDL, the DialectOp has a name attribute for the dialect name).

I can think of two ways:

a) Rename `name` to `_name`. This is what Python usually does for "already used" names in functions etc.

b) The main issue comes from storing both the name of the operation and its attributes in the same way. We could also implement a dictionary mapping classes to their names. This would probably live within the MLContext and it would mean that printer and parser would need to access it and map things around. Obviously in the case of parsing we would need the reverse mapping (string to Class).

Any ideas/inputs?

**Comment by superlopuh** [comment]
`op_name` etc? Could it be passed as a parameter to the class annotation `@irdl_attr_definition(name="dense")`?

**Reply by superlopuh** [reply]
Ah I see both of these were already suggested in the PR :)

**Reply by webmiche** [reply]
I think it can be passed in, but that still leaves the question where to store it.

I would combine my second suggestions with passing the name into the decorator.
